AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/StateMachine.swift

```diff
@@ -36,12 +39,28 @@ class StateMachine<State, Action> {
     /// state will be published to `state`, before the `notify` action returns.
     func notify(action: Action) {
         queue.sync {
-            log.verbose("Notifying: \(action)")
+            self.log.verbose("Notifying: \(action)")
             let newState = self.resolve(currentState: self.state, action: action)
             self.state = newState
         }
     }
 
+    /// - Warning: This method is not thread safe because it will resolve the new state and
+    /// apply update the `state` to the new state in a non-atomic way.
+    ///
+    /// Call this method when you want to perform the state machine's `responder` closure with the new state resolved
+    /// from the current state and action.
+    func respond(action: Action) {
+        guard let responder = responder else {
+            log.error("Use of `StateMachine.respond` without an attached responder is a programming error.")
+            return
+        }
+        log.verbose("Responding: \(action)")
+        let newState = resolve(currentState: state, action: action)
```

Comment:
maybe i can make this thread safe using a DispatchSemaphore, but acquiring and releasing a semaphore in a single ReconileAndSaveOperation would occur for each `stateMachine.respond` call, which is about 5 or so calls